cmd/maya-exporter/app/collector/zvol/pool_synctime.go

```diff
@@ -0,0 +1,137 @@
+// Copyright © 2017-2019 The OpenEBS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zvol
+
+import (
+	"os"
+	"strings"
+	"sync"
+
+	col "github.com/openebs/maya/cmd/maya-exporter/app/collector"
+	types "github.com/openebs/maya/pkg/exec"
+	zvol "github.com/openebs/maya/pkg/zvol/v1alpha1"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/klog"
+)
+
+// poolMetrics implements prometheus.Collector interface
+type poolMetrics struct {
+	sync.Mutex
+	*poolSyncMetrics
+	request bool
+	runner  types.Runner
+}
+
+// NewPoolSyncMetric returns new instance of poolMetrics
+func NewPoolSyncMetric(runner types.Runner) col.Collector {
+	return &poolMetrics{
+		poolSyncMetrics: newPoolMetrics().
+			withZpoolLastSyncTime().
+			withZpoolStateUnknown().
+			withzpoolLastSyncTimeCommandError(),
+		runner: runner,
+	}
+}
+
+func (p *poolMetrics) setRequestToFalse() {
+	p.Lock()
+	p.request = false
+	p.Unlock()
+}
+
+// collectors returns the list of the collectors
+func (p *poolMetrics) collectors() []prometheus.Collector {
+	return []prometheus.Collector{
+		p.zpoolLastSyncTime,
+		p.zpoolStateUnknown,
+		p.zpoolLastSyncTimeCommandError,
+	}
+}
+
+// Describe is implementation of Describe method of prometheus.Collector
+// interface.
+func (p *poolMetrics) Describe(ch chan<- *prometheus.Desc) {
+	for _, col := range p.collectors() {
+		col.Describe(ch)
+	}
+}
+
+func (p *poolMetrics) checkError(stdout []byte) *poolfields {
+
+	if zvol.IsNoDataSetAvailable(string(stdout)) || IsNoPoolAvailable(string(stdout)) {
+		pool := poolfields{
+			name:                          os.Getenv("HOSTNAME"),
+			zpoolLastSyncTime:             0,
+			zpoolStateUnknown:             1,
+			zpoolLastSyncTimeCommandError: 0,
+		}
```

Comment:
Can you please tell me which is the good place to define constants. Should i define it in zpool.go where all the constants related to pool is defined or In the same file as its just 0 and 1 so it can be in the same file.

Reply:
https://github.com/openebs/maya/blob/master/pkg/zpool/v1alpha1/zpool.go#L50-L61. Here also we are not using constant. So, can you tell me where to use it and where not, what is the best practice, it will be my learning.